Basic repo setup, and import inetdiag utilities.

gfr10598 opened the pull request:
1.  Set up travis
2.  Add badges to README.md
3.  Import inetdiag utility code and adapt for our use.
4.  Add some unit tests and get coverage to reasonable level.

gfr10598 commented:
Issue #2

pboothe commented:
Review status: 0 of 6 files reviewed at latest revision, all discussions resolved.

---

*[.travis.yml, line 64 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-IH8E1Cjac59UjeMk:-LC-IH8E1Cjac59UjeMl:bm46rti) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/.travis.yml#L64)):*
> ```YAML
> # the non-integration tests are repeated.  If we change the unit tests to NOT run when integration
> # test tag is set, then we would need to have separate cov files.
> # Note: we do not run integration tests from forked PRs b/c the SA is unavailable.
> ```

"from forked repos" not "from forked PRs"

---

*[inetdiag/inetdiag.go, line 38 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-IZE91nH-1LMfXZPa:-LC-IZE91nH-1LMfXZPb:bopllbm) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L38)):*
> ```Go
> // netinet/tcp.h
> const (
> 	_               = iota
> ```

Won't this cause lint warnings because local variable `_` is shadowing global variable `_`? Or is go smart enough to realize that `_` is a trash variable name?

---

*[inetdiag/inetdiag.go, line 56 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-JC6mB5edHjTwtReG:-LC-JC6mB5edHjTwtReH:b-gz22pp) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L56)):*
> ```Go
> )
> 
> var tcpStatesMap = map[uint8]string{
> ```

https://godoc.org/golang.org/x/tools/cmd/stringer ?

---

*[inetdiag/inetdiag.go, line 96 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-J_xvBJxq8k6QmOda:-LC-J_xvBJxq8k6QmOdb:b9qbjn9) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L96)):*
> ```Go
> }
> 
> // TODO should use more net.IP code instead of custom code.
> ```

Make issue, link to issue :)

---

*[inetdiag/inetdiag.go, line 105 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-KgOR2tXr5wdZEpuW:-LC-KgOR2tXr5wdZEpuX:bxaaeud) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L105)):*
> ```Go
> }
> 
> func isIpv6(original [16]byte) bool {
> ```

Let's just hope we never get a very-improbable v6 address like AB:CD:EF:12:00:00:00:00:00:00:00:00:00:00:00:00 ?

---

*[inetdiag/inetdiag.go, line 126 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-LBwS3JxK8OZoIdYQ:-LC-LBwS3JxK8OZoIdYR:b-7e2p7b) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L126)):*
> ```Go
> }
> 
> // InetDiagReqV2 is the Netlink request struct.
> ```

as defined in...

---

*[inetdiag/inetdiag.go, line 161 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-JkKvB1G1mhP71lCj:-LC-JkKvB1G1mhP71lCk:bgx06i9) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L161)):*
> ```Go
> 
> // InetDiagMsg is the linux binary representation of a InetDiag message header.  Note that this
> // does NOT take into account byte ordering.
> ```

I don't understand whether this means I should correct the byte ordering before I put things in to this struct or after I get them out or just ignore the issue altogether.

---

*[inetdiag/inetdiag_test.go, line 122 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-KTtd6KfqjbL5f67u:-LC-KTtd6KfqjbL5f67v:brfmurn) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag_test.go#L122)):*
> ```Go
> 	hdr, _ := inetdiag.ParseInetDiagMsg(data[:])
> 
> 	if hdr.ID.SrcIP().IsLoopback() {
> ```

It seems like there are more conditions being tested here than just "is it loopback" ?

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/tcp-info/6)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
Review status: 0 of 6 files reviewed at latest revision, 8 unresolved discussions.

---

*[.travis.yml, line 64 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-IH8E1Cjac59UjeMk:-LC-MVbhDMwkbnw3rbfI:btxav73) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/.travis.yml#L64)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

"from forked repos" not "from forked PRs"
</blockquote></details>

Wow - you actually read this?  Copied from other travis files, so I don't even see the old comments.  Thanks.

---

*[inetdiag/inetdiag.go, line 38 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-IZE91nH-1LMfXZPa:-LC-MkTn2WFrT81hWpzW:blrj4l7) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L38)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

Won't this cause lint warnings because local variable `_` is shadowing global variable `_`? Or is go smart enough to realize that `_` is a trash variable name?
</blockquote></details>

_ is go syntax for an unused value.  I don't think you can actually reference it.

---

*[inetdiag/inetdiag.go, line 56 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-JC6mB5edHjTwtReG:-LC-NO6_4ykRcosgD70J:b3nr534) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L56)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

https://godoc.org/golang.org/x/tools/cmd/stringer ?
</blockquote></details>

Interesting tool.  Will keep it in mind.

---

*[inetdiag/inetdiag.go, line 96 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-J_xvBJxq8k6QmOda:-LC-NWjl1e9Bd5Syizmb:b-s5scr0) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L96)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

Make issue, link to issue :)
</blockquote></details>

I don't think it merits its own issue.  I would rate it p4 and we would never look at it.

---

*[inetdiag/inetdiag.go, line 105 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-KgOR2tXr5wdZEpuW:-LC-Ne30-ZFrI-oQ3HD7:b-nyw5qv) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L105)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

Let's just hope we never get a very-improbable v6 address like AB:CD:EF:12:00:00:00:00:00:00:00:00:00:00:00:00 ?
</blockquote></details>

This is consistent with how the net.IP library works, so I'm not too worried.
8-)

---

*[inetdiag/inetdiag.go, line 126 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-LBwS3JxK8OZoIdYQ:-LC-_omL-763Hf07xvbR:b-896fix) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L126)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

as defined in...
</blockquote></details>

Done.

---

*[inetdiag/inetdiag.go, line 161 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-JkKvB1G1mhP71lCj:-LC-b1jN9lfsZvxdIIga:b-5h66cb) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L161)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

I don't understand whether this means I should correct the byte ordering before I put things in to this struct or after I get them out or just ignore the issue altogether.
</blockquote></details>

Oops.  I added that earlier and discovered it is misleading.  Fixed.

---

*[inetdiag/inetdiag_test.go, line 122 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-KTtd6KfqjbL5f67u:-LC-fvk52JnIFFVLtmht:b-tstmu4) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag_test.go#L122)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

It seems like there are more conditions being tested here than just "is it loopback" ?
</blockquote></details>

Maybe you were looking at an older version?  Not sure how that would have happened, but I did have some other stuff in earlier commits.

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/tcp-info/6)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
PTAL.

---

Review status: 0 of 6 files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/tcp-info/6#-:-LC-gKL7EsZf8ZJ1kW22:b-83330m)*
<!-- Sent from Reviewable.io -->

pboothe commented:
Review status: 0 of 6 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[.travis.yml, line 64 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-IH8E1Cjac59UjeMk:-LC-jsN_3aBpOu8gtC7b:bwgeeqi) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/.travis.yml#L64)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Wow - you actually read this?  Copied from other travis files, so I don't even see the old comments.  Thanks.
</blockquote></details>

But the text is unchanged?

---

*[inetdiag/inetdiag.go, line 96 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-J_xvBJxq8k6QmOda:-LC-jVth7uIE7SjZeWIp:b-of1pfe) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L96)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

I don't think it merits its own issue.  I would rate it p4 and we would never look at it.
</blockquote></details>

But if you make it an issue, then we can ask someone else to fix it. :)

---

*[inetdiag/inetdiag_test.go, line 122 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-KTtd6KfqjbL5f67u:-LC-iv9O6g1_LYdVJnGa:b-nefwm0) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag_test.go#L122)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Maybe you were looking at an older version?  Not sure how that would have happened, but I did have some other stuff in earlier commits.
</blockquote></details>

I was unclear and too brief.

The function TestID6 does a whole lot of arranging bytes just so. However, the only assertions that the test makes (as near as I can tell) are "Do `hdr.ID.SrcIP().IsLoopback()` and  `hdr.ID.DstIP().IsLoopback()` both return false?"

The amount of boilerplate and setup seems out of place compared to the rigorousness of the tests. Are more than those two conditions being tested? If so, where? If not, should they be?

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/tcp-info/6)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
Review status: 0 of 6 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[.travis.yml, line 64 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-IH8E1Cjac59UjeMk:-LC-txGbCU81zJ0vDKRw:b-2108dt) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/.travis.yml#L64)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

But the text is unchanged?
</blockquote></details>

Oops.  Got distracted.

---

*[inetdiag/inetdiag.go, line 96 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-J_xvBJxq8k6QmOda:-LC-u3vnBnkSlZkNpjU1:bru0rv5) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag.go#L96)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

But if you make it an issue, then we can ask someone else to fix it. :)
</blockquote></details>

grep TODO...  8-)

---

*[inetdiag/inetdiag_test.go, line 122 at r1](https://reviewable.io/reviews/m-lab/tcp-info/6#-LC-KTtd6KfqjbL5f67u:-LC-umfqA5EXUbGdrw6i:b-19vs0r) ([raw file](https://github.com/m-lab/tcp-info/blob/1e949cafd5a6c682f887463248cca6636470b212/inetdiag/inetdiag_test.go#L122)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

I was unclear and too brief.

The function TestID6 does a whole lot of arranging bytes just so. However, the only assertions that the test makes (as near as I can tell) are "Do `hdr.ID.SrcIP().IsLoopback()` and  `hdr.ID.DstIP().IsLoopback()` both return false?"

The amount of boilerplate and setup seems out of place compared to the rigorousness of the tests. Are more than those two conditions being tested? If so, where? If not, should they be?
</blockquote></details>

Got it.  Cleaned it up.  Thanks.

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/tcp-info/6)*
<!-- Sent from Reviewable.io -->
